documentation/markdown/architecture/overview.md

```diff
@@ -0,0 +1,68 @@
+Many things have been added since the original inception of the project,
+but the core ideas within that document are still valid.
+
+As can be seen from the architecture, an important idea is the modularity of all components.
+No actual implementations are defined there, only their interfaces.
+Making all the components independent of each other in such a way provides us with an enormous flexibility:
+they can all be replaced by a different implementation, without impacting anything else.
+This is how we can provide many different configurations for the server,
+and why it is impossible to provide ready solutions for all possible combinations.
+
+## Architecture diagrams
+
+Having a modular architecture makes it more difficult to give a complete architecture overview.
+We will limit ourselves to the more commonly used default configurations we provide,
+and in certain cases we might give examples of what differences there are
+based on what configurations are being imported.
+
+To do this we will make use of architecture diagrams.
+We will use an example below to explain the formatting used throughout the architecture documentation:
+
+```mermaid
+flowchart TD
+  LdpHandler("<strong>LdpHandler</strong><br>ParsingHttphandler")
+  LdpHandler --> LdpHandlerArgs
+  
+  subgraph LdpHandlerArgs[" "]
+    RequestParser("<strong>RequestParser</strong><br>BasicRequestParser")
+    Auth("<br>AuthorizingHttpHandler")
+    ErrorHandler("<strong>ErrorHandler</strong><br><i>ErrorHandler</>")
+    ResponseWriter("<strong>ResponseWriter</strong><br>BasicResponseWriter")
+  end
+```
+
+Every square represents a component that is instantiated in the Components.js [configuration](dependency-injection.md).
+
+The first (bold) line is a shorthand for the identifier that component has in the configuration.
+In this example, **LdpHandler** is a shorthand for the actual identifier `urn:solid-server:default:LdpHandler`,
+which you can find in the relevant configuration files.
+In case the full URI is not specified, 
+it can usually be found by prepending `urn:solid-server:default:` to the shorthand identifier.
+
+The second line is the class this is an instantiation of, so in this case a `ParsingHttpHandler`.
+The arrow points to the input parameters this class gets, 4 in this case,
+which are instanced components themselves.
+
+If the order of the parameters matters, such as in the case of a [`WaterfallHandler`](core.md), 
+arrows will be added to indicate this.
+
+As can be seen, some instances do not have an identifier in the diagram.
+This means these do not have a unique identifier in the configuration
+and you should look for the parent in the configurations if you want to find them.
+
+If the class name is in cursive,
+this means that this is an interface and the actual class
+is not relevant for the explanation being given,
+and might even differ depending on which configuration is being used.
```

Comment:
I could follow the above, but it didn't work well as a reference / legend because I couldn't scan it.

Perhaps consider making it more itemized:

- Box
  - First line
    - **Bold text**: shorthand for the actual identifier
    - (empty): this component has no unique identifier, use the parent to…
  - Second line
    - Regular text: Instantiated class name
    - _Italic text: Implemented interface name
- Arrow
  - Indicates…

For example, in the above:
- Box
 - **LdpHandler** stands for `urn:solid-server:default:LdpHandler`
 - …

Reply:
Would the following be preferable? Would replace all the text below the example diagram

> Below is a summary of how to interpret such diagrams:
> 
> * Rounded red box: component instantiated in the Components.js [configuration](dependency-injection.md).
>   * First line: 
>     * **Bold text**: shorthand of the instance identifier. In case the full URI is not specified,
>       it can usually be found by prepending `urn:solid-server:default:` to the shorthand identifier.
>     * (empty): this instance has no identifier and is defined in the same place as its parent.
>   * Second line: 
>     * Regular text: The class of this instance.
>     * _Italic text_: The interface of this instance. 
>       Will be used if the actual class is not relevant for the explanation or can differ.
> * Square grey box: the parameters of the linked instance.
> * Arrow: links an instance to its parameters. Can also be used to indicate the order of parameters if relevant.
> 
> For example, in the above, **LdpHandler** is a shorthand for the actual identifier `urn:solid-server:default:LdpHandler`
> and is an instance of `ParsingHttpHandler`. It has 4 parameters,
> one of which has no identifier but is an instance of `AuthorizingHttpHandler`.

Reply:
> * Rounded red box:

Haha I was going to say, but it's not red though. Then I realize I'm in dark mode and I had a totally different color.

Yeah, looks great, this is very scannable. Makes it easy to keep that page open as a legend for the others.